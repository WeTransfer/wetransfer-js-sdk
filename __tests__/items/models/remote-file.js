const { RemoteFile } = require('../../../src/items/models');

describe('RemoteFile model', () => {
  let file = null;
  beforeEach(() => {
    file = new RemoteFile({
      id: 'random-hash',
      content_identifier: 'file',
      local_identifier: 'delightful-cat',
      meta: {
        multipart_parts: 3,
        multipart_upload_id: 'some.random-id--'
      },
      name: 'kittie.gif',
      size: 195906,
      upload_id: 'more.random-ids--',
      upload_expires_at: 1520410633
    });
  });

  describe('contructor', () => {
    it('should create a normalized file', () => {
      expect(file).toMatchSnapshot();
    });
  });
});
