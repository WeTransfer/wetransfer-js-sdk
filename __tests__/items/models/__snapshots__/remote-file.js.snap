// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoteFile model contructor should create a normalized file 1`] = `
RemoteFile {
  "content_identifier": "file",
  "id": "random-hash",
  "local_identifier": "delightful-cat",
  "meta": Object {
    "multipart_parts": 3,
    "multipart_upload_id": "some.random-id--",
  },
  "name": "kittie.gif",
  "size": 195906,
}
`;
